libmamba/tests/src/doctest-printer/pair.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <utility>
+
+#include <doctest/doctest.h>
+#include <fmt/format.h>
+
+namespace doctest
+{
+    template <typename T, typename U>
+    struct StringMaker<std::pair<T, U>>
+    {
+        static auto convert(const std::pair<T, U>& value) -> String
+        {
+            return { fmt::format("std::pair{{{}, {}}}", value.first, value.second).c_str() };
```

Comment:
Why do we convert to a c-string in the middle, to have it null-terminated?
A more general question, do you think there will be a lot of these functions so that it could be more readable to put them in separate files, or maybe the opposite? Because otherwise we could also put everything in one file.

Reply:
Well, doctest does not use `std::string` anywhere, not even as conversion...
They have a `String` that can be built from a `const char*`, so I pass it the `c_str()` 

I think more would come yes. Also it is to avoid including all of `<map>` `<set>`, `<tuple>`, `"graph.hpp"` in every test file.